main/src/library/LazyList.flix

```diff
@@ -398,4 +398,9 @@ namespace LazyList {
         case LCons(x, xs) => f(x); foreach(f, force xs)
         case LList(xs)    => foreach(f, force xs)
     }
+
+    ///
+    /// Returns an infinite LazyList of repeating `x`s.
+    ///
+    pub def repeat(x: a): LazyList[a] = LList(lazy LCons(x, lazy repeat(x)))
```

Comment:
I think this will introduce an extra LList at every level. It would be better to use a repeatHelper function.

Reply:
Good idea. I know that we want to avoid evaluating the first element but in this case all elements are the same so is it important that we return an `Llist`?

Reply:
In the recursive call, don't call `lazy repeat`, but `lazy repeatHelper` which does not wrap in LLlist.

Reply:
Yeah I follow your idea but I was wondering whether we can skip the initial `Llist` step and simply return an `LCons` immediately such that no helper function is required.

Reply:
I.e.
```scala
pub def repeat(x: a): LazyList[a] = LCons(x, lazy repeat(x))
```

Reply:
Of course the first element will be evaluated but since repeat will always generate `x` would it make sense to allow the evaluation of the first element in this special case?

Reply:
> 
> 
> I.e.
> 
> ```scala
> pub def repeat(x: a): LazyList[a] = LCons(x, lazy repeat(x))
> ```

I think this is a pragmatic choice.